packages/teams-js/test/public/pages.spec.ts

```diff
@@ -2248,6 +2280,101 @@ describe('Testing pages module', () => {
             expect(pages.navigateCrossDomain('https://valid.origin.com')).rejects.toEqual(errorNotSupportedOnPlatform);
           });
 
+          it(`pages.navigateCrossDomain should throw error with a bad origin when initialized with ${context}`, async () => {
+            await framelessPostMocks.initializeWithContext(context);
+            const promise = pages.navigateCrossDomain('https://badorigin.com');
+            const navigateCrossDomainMessage = framelessPostMocks.findMessageByFunc('navigateCrossDomain');
+            expect(navigateCrossDomainMessage).not.toBeNull();
+            framelessPostMocks.respondToMessage({
```

Comment:
Are these testing teams-js validation logic, or are they validating that these calls throw correctly when we receive an error response?

Reply:
Yes we are checking the URL validation logic

Reply:
If you go and check the HUB SDK we already have it implemented